src/Microsoft.FeatureManagement/IFilterParametersBinder.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.FeatureManagement
+{
+    /// <summary>
+    /// An interface used by the feature management system to pre-bind feature filter parameters to a settings type.
+    /// <see cref="IFeatureFilter"/>s can implement this interface to take advantage of caching of settings by the feature management system.
+    /// </summary>
+    public interface IFilterParametersBinder
```

Comment:
We are caching the parameters, not the whole feature definition. I'm wondering if it makes sense to name this interface something like `ICacheableFeatureFilterParameters`, and the requirement is that the `BindParameters` returns a new object instance every call. This way, we don't need the `IFeatureDefinitionProviderCacheable`. In your current code, implementation of this interface is a prerequisite for caching anyway. Thoughts?

Reply:
Not sure I follow. This is a new interface, isn't it? Typical feature filters implement `IFeatureFilter`, not this one, right?

Reply:
Correct- so essentially:

A **filter** will not use caching, unless the filter implements this interface. 

Meaning the TimeWindowFilter that used to be:
```
public class TimeWindowFilter : IFeatureFilter
```
Has and will continue to be a valid filter. However we want this filter to handle caching, so it becomes:
```
public class TimeWindowFilter : IFeatureFilter, IFilterParametersBinder
```

And then separately, we also added a private interface for the **Provider**. So our provider declares if it can handle caching. 

If the Provider can handle caching, it will use it, but only for Filters that can handle caching (aka- implement this interface)

Reply:
Right. I don't see the need for a provider to declare cacheable at this point. We have agreed not to expose the cacheability to custom filters in this version yet.

Reply:
No, we agreed not to expose cacheable to custom providers, as they're the location we can't guarantee a ref update. Custom Filters don't have the same risk.

Reply:
What's the customer value of exposing this interface without cacheability?

Reply:
If they are using our default provider- they can use Custom Filters with caching enabled.

Reply:
Thanks for the clarification. Is this interface useful in scenarios other than caching?

Reply:
Not that I can think of. Only other benefit is a well-defined way to go from IConfiguration to the filter specific CustomFilterSettings object. But I don't think that it would be used for that.

Reply:
Right. It also appears to me this interface is only useful if a filter binds parameters to an object and cares about caching. A custom filter can work directly with the `IConfiguration` of parameters without worrying about all this. Can we name the interface so it's clearer why a user wants to implement it?

Reply:
The interface exposes a single method, `BindParameters`. I'd argue `IFilterParametersBinder` is aptly named. Any additional information about usage seems better placed in a summary.

Reply:
After re-reading the summaries- I can't come up with a better description that provides any extra info. I'm going to resolve this conversation for now- but let me know if you can come up with a better summary. 

Pasting the current summaries here for easy reference:
```
    /// <summary>
    /// An interface used by the feature management system to pre-bind feature filter parameters to a settings type.
    /// <see cref="IFeatureFilter"/>s can implement this interface to take advantage of caching of settings by the feature management system.
    /// </summary>
    public interface IFilterParametersBinder
    {
        /// <summary>
        /// Binds a set of feature filter parameters to a settings object.
        /// </summary>
        /// <param name="parameters">The configuration representing filter parameters to bind to a settings object</param>
        /// <returns>A settings object that is understood by the implementer of <see cref="IFilterParametersBinder"/>.</returns>
        object BindParameters(IConfiguration parameters);
    }
```

Reply:
@zhenlan are you good with this?

Reply:
@rossgrambo 

Discussed offline with @zhenlan. We agreed to keep the interface the way it is now.